crate/server/src/core/cover_crypt/rekey_keys.rs

```diff
@@ -31,59 +34,157 @@ use crate::{
+                owner,
+                params,
+            )
+            .await?;
+
+            (msk_uid, mpk_uid)
+        }
+        RekeyEditAction::RemoveAttribute(attrs) => {
+            let (msk_uid, msk_object, mpk_uid) =
+                update_master_keys(kmip_server, owner, params, msk_uid, |policy, msk, mpk| {
+                    attrs
+                        .iter()
+                        .try_for_each(|attr| policy.remove_attribute(attr))?;
+                    cover_crypt.update_master_keys(policy, msk, mpk)?;
+                    Ok(())
+                })
+                .await?;
+
+            update_user_secret_keys(
+                kmip_server,
+                cover_crypt,
+                &msk_uid,
+                &msk_object,
+                owner,
+                params,
+            )
+            .await?;
+
+            (msk_uid, mpk_uid)
+        }
+        RekeyEditAction::DisableAttribute(attrs) => {
+            let (msk_uid, _, mpk_uid) =
+                update_master_keys(kmip_server, owner, params, msk_uid, |policy, msk, mpk| {
+                    attrs
+                        .iter()
+                        .try_for_each(|attr| policy.disable_attribute(attr))?;
+                    cover_crypt.update_master_keys(policy, msk, mpk)?;
+                    Ok(())
+                })
+                .await?;
+
+            (msk_uid, mpk_uid)
+        }
+        RekeyEditAction::RenameAttribute(pairs_attr_name) => {
+            let (msk_uid, _, mpk_uid) =
+                update_master_keys(kmip_server, owner, params, msk_uid, |policy, msk, mpk| {
+                    pairs_attr_name.iter().try_for_each(|(attr, new_name)| {
+                        policy.rename_attribute(attr, new_name.clone())
+                    })?;
+                    cover_crypt.update_master_keys(policy, msk, mpk)?;
+                    Ok(())
+                })
+                .await?;
+
+            (msk_uid, mpk_uid)
+        }
+        RekeyEditAction::AddAttribute(attrs_properties) => {
+            let (msk_uid, _, mpk_uid) =
+                update_master_keys(kmip_server, owner, params, msk_uid, |policy, msk, mpk| {
+                    attrs_properties
+                        .iter()
+                        .try_for_each(|(attr, encryption_hint)| {
+                            policy.add_attribute(attr.clone(), *encryption_hint)
+                        })?;
+                    cover_crypt.update_master_keys(policy, msk, mpk)?;
+                    Ok(())
+                })
+                .await?;
+
+            (msk_uid, mpk_uid)
+        }
+    };
+
+    Ok(ReKeyKeyPairResponse {
+        private_key_unique_identifier: UniqueIdentifier::TextString(msk_uid.to_string()),
+        public_key_unique_identifier: UniqueIdentifier::TextString(mpk_uid),
+    })
+}
+
+/// Update the master key with a new Policy
+/// (after editing the policy typically)
+pub async fn update_master_keys(
```

Comment:
The function name and docstring are no longer really accurate since this method will update the policy, master keys and even the user keys if needed

Reply:
Good catch, I changed the documentation.
I kept the name since this function actually updates the master keys. Since the policy will soon be part of the MSK, I did not want to reflect it in the naming. And the USK are not updated by this function: the dedicated `update_user_secret_keys` procedure is in charge of this.